maltoolbox/attackgraph/attackgraph.py

```diff
@@ -493,6 +333,176 @@ def attach_attackers(self) -> None:
 
             attacker.entry_points = list(attacker.reached_attack_steps)
 
+    def _follow_expr_chain(
+            self,
+            model: Model,
+            target_assets: list[Any],
+            expr_chain: ExpressionsChain
+        ) -> list:
+        """
+        Recursively follow a language graph expressions chain on an instance
+        model.
+
+        Arguments:
+        model           - a maltoolbox.model.Model on which to follow the
+                          expressions chain
+        target_assets   - the list of assets that this expressions chain
+                          should apply to. Initially it will contain the
+                          asset to which the attack step belongs
+        expr_chain      - the expressions chain we are following
+
+        Return:
+        A list of all of the target assets.
+        """
+
+        if expr_chain is None:
+            # This is an attack step targeting the same assets
+            return target_assets
+
+        if logger.isEnabledFor(logging.DEBUG):
+            # Avoid running json.dumps when not in debug
+            logger.debug(
+                'Following Expressions Chain:\n%s',
+                json.dumps(expr_chain.to_dict(), indent = 2)
+            )
+
+        match (expr_chain.type):
+            case 'union' | 'intersection' | 'difference':
+                # The set operators are used to combine the left hand and
+                # right hand targets accordingly.
+                if not expr_chain.left_link:
+                    raise LanguageGraphException('"%s" step expression chain'
+                        ' is missing the left link.' % expr_chain.type)
+                if not expr_chain.right_link:
+                    raise LanguageGraphException('"%s" step expression chain'
+                        ' is missing the right link.' % expr_chain.type)
```

Comment:
If this is the case and there is indeed such an error, is the underlying MAL code valid? If not, this should be checked for at the compiler level.

Reply:
These were added because mypy would complain about it.